File: clients/csharp/src/Models/Blob.cs
namespace MyMind.Models;

/// <summary>
/// A binary payload — used both for uploads (on <see cref="MyMind.Actions.CreateObjectRequest"/>)
/// and for downloads (returned by <see cref="MyMind.Services.ObjectsService.DownloadAsync"/>).
/// </summary>
public class Blob
{
    public required Stream Stream { get; init; }
    public required string Type { get; init; }
    public string? Name { get; init; }
}

File: clients/swift/src/Models/Blob.swift
import Foundation

/// A binary payload — used both for uploads (on `CreateObjectRequest`)
/// and for downloads (returned by `ObjectsService.download`).
public struct Blob: Sendable {
    public let data: Data
    public let type: String
    public let name: String?

    public init(data: Data, type: String, name: String? = nil) {
        self.data = data
        self.type = type
        self.name = name
    }
}
